frontend/src/workStation/CodeGroup.jsx

```diff
@@ -0,0 +1,352 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+import Nav from "../app/Nav";
+import Footer from "../app/Footer";
+import { useParams, useNavigate } from "react-router-dom";
+import {
+  Box,
+  Button,
+  HStack,
+  useDisclosure,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+} from "@chakra-ui/react";
+
+function CodeGroup() {
+  const [groupData, setGroupdata] = useState([]);
+  const [allFiles, setAllFiles] = useState([]);
+  const [allMembers, setAllMembers] = useState([]);
+  const [newGroupName, setNewGroupName] = useState("");
+  const [newGroupImageUrl, setNewGroupImageUrl] = useState("");
+  const [newMemberInput, setNewMemberInput] = useState("");
+  const [newMembers, setNewMembers] = useState([]);
+  const [newFileName, setNewFileName] = useState("");
+  const [error, setError] = useState("");
+  const {
+    isOpen: isUpdateGroupModalOpen,
+    onOpen: onUpdateGroupModalOpen,
+    onClose: onUpdateGroupModalClose,
+  } = useDisclosure();
+  const {
+    isOpen: isAddMembersModalOpen,
+    onOpen: onAddMembersModalOpen,
+    onClose: onAddMembersModalClose,
+  } = useDisclosure();
+  const {
+    isOpen: isCreateNewFileModalOpen,
+    onOpen: onCreateNewFileModalOpen,
+    onClose: onCreateNewFileModalClose,
+  } = useDisclosure();
+
+  const { groupId } = useParams();
+  const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+  const token = localStorage.getItem("jwt");
+  const navigate = useNavigate();
+  const options = {
```

Comment:
Nice job making this a variable so you don't run into the authentication issue you were having again

Reply:
Thanks!